internal/factsengine/gatherers/passwd.go

```diff
@@ -0,0 +1,137 @@
+package gatherers
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+)
+
+const (
+	PasswdGathererName = "passwd"
+	PasswdFilePath     = "/etc/passwd"
+)
+
+// nolint:gochecknoglobals
+var (
+	PasswdFileError = entities.FactGatheringError{
+		Type:    "passwd-file-error",
+		Message: "error reading /etc/passwd file",
+	}
+
+	PasswdDecodingError = entities.FactGatheringError{
+		Type:    "passwd-decoding-error",
+		Message: "error decoding file content",
+	}
+)
+
+// An Entry contains all the fields for a specific user
+type Entry struct {
+	User        string `json:"user"`
+	UID         string `json:"uid"`
+	GID         string `json:"gid"`
+	Description string `json:"description"`
+	Home        string `json:"home"`
+	Shell       string `json:"shell"`
+}
+
+type PasswdGatherer struct {
+	passwdFilePath string
+}
+
+func NewDefaultPasswdGatherer() *PasswdGatherer {
+	return NewPasswdGatherer(PasswdFilePath)
+}
+
+func NewPasswdGatherer(path string) *PasswdGatherer {
+	return &PasswdGatherer{
+		passwdFilePath: path,
+	}
+}
+
+func (g *PasswdGatherer) Gather(factsRequests []entities.FactRequest) ([]entities.Fact, error) {
+	facts := []entities.Fact{}
+	log.Infof("Starting %s state facts gathering process", PasswdGathererName)
```

Comment:
What does the word `state` reflect here? I'd use "file"

Reply:
arggg
copy/paste typo